tests/runner/test_dataset_block.py

```diff
@@ -131,107 +169,167 @@ def test_partial_block_for_chunked_data(
 def test_partial_block_for_chunked_data_with_padding_center(
     slicing_dim: Literal[0, 1, 2]
 ):
-    block_shape = [10, 10, 10]
-    block_shape[slicing_dim] = 6
-    start_index = 3
-    data = np.ones(block_shape, dtype=np.float32)
-    global_index = [0, 0, 0]
-    global_index[slicing_dim] = 10 + start_index
+    # sizes and shapes
     global_shape_t = [10, 10, 10]
     global_shape_t[slicing_dim] = 30
     global_shape = make_3d_shape_from_shape(global_shape_t)
-    chunk_shape_t = [10, 10, 10]
-    chunk_shape_t[slicing_dim] += 4  # for padding
-    chunk_shape = make_3d_shape_from_shape(chunk_shape_t)
+
+    unpadded_chunk_shape = (10, 10, 10)
+    padding = (2, 2)
+    padded_chunk_shape_t = list(unpadded_chunk_shape)
+    padded_chunk_shape_t[slicing_dim] += padding[0] + padding[1]
+    padded_chunk_shape = make_3d_shape_from_shape(padded_chunk_shape_t)
+
+    unpadded_block_slices = 2
+    unpadded_block_shape_t = list(unpadded_chunk_shape)
+    unpadded_block_shape_t[slicing_dim] = unpadded_block_slices
+    unpadded_block_shape = make_3d_shape_from_shape(unpadded_block_shape_t)
+    padded_block_shape_t = unpadded_block_shape_t
+    padded_block_shape_t[slicing_dim] += padding[0] + padding[1]
+    padded_block_shape = make_3d_shape_from_shape(padded_block_shape_t)
+    block_start_index = 3
```

Comment:
Firstly, is the code comment I've suggested correct?

```suggestion
    block_start_index = 3  # start of block within chunk, without padding
```

If so, this is just for the sake of clarifying things, because I got confused initially why `block_start_index` was being shifted by the "before" padding value later in the test.

Reply:
The comment is not correct here - it's including the padding. It's the start index of the block's data, incl. padding. That's reflected in the later assert: 

```python
assert block.chunk_index[slicing_dim] == block_start_index
```

since the block's properties for `chunk_index` are including the padded slices. It's the index of the data in the block within the chunk, and the data includes the padded slices.

Reply:
Gotcha, I wasn't at all confident about the comment I suggested, thanks for clarifying.